src/nomad_simulation_parsers/parsers/h5md/parser.py

```diff
@@ -0,0 +1,358 @@
+        step_data = {}
+        value = self.get_value('value', data)
+        steps = self.get_value('step', data)
+        if value is None or steps is None:
+            return step_data
+        index = steps.index(step)
+        step_data['value'] = value[index]
+        times = self.get_value('time', data)
+        step_data['time'] = times[index]
+        return step_data
+
+    def get_cell_data(self, source: dict[str, Any]) -> dict[str, Any]:
+        if not source.get('step'):
+            return {}
+        particles = self.data.get('particles', {}).get('all')
+        if particles is None:
+            return {}
+
+        source_paths = [
+            ('lattice_vectors', 'box.edges'),
+        ]
+        system_data = {}
+        for name, path in source_paths:
+            data = self.get_source(particles, path)
+            if not data:
+                continue
+            step_data = self.get_step_data(data, source.get('step'))
+            if not step_data:
+                continue
+            system_data.setdefault('time', step_data['time'])
+            if step_data['time'] == system_data['time']:
+                system_data[name] = step_data['value']
+        box = particles.get('box', {})
+        system_data['boundary'] = box.get(f'{self.attribute_prefix}boundary')
+
+        return system_data
+
+    def to_species_labels(
+        self, source: dict[str, Any], **kwargs
+    ) -> list[dict[str, Any]]:
+        if source.get('step') is None:
+            return []
+
+        source_data = self.get_source(self.data, kwargs['path'])
+
+        return [{'chemical_symbol': s, 'label': s} for s in source_data]
+
+    def get_top_system_quantity(
+        self, source: dict[str, Any], **kwargs
+    ) -> list[dict[str, Any]]:
+        if source.get('step') is None:
+            return []
+
+        source_data = self.get_source(self.data, kwargs['path'])
+        if kwargs.get('func'):
+            try:
+                source_data = kwargs.get('func')(source_data)
+            except Exception:
+                self.logger.warning(
+                    'Error applying function to get top level system quantity, '
+                    'will not be populated.'
+                )
+                return []
+
+        return self.get_source(self.data, kwargs['path'])
+
+    def get_output_steps(self, source: dict[str, Any]) -> list[dict[str, Any]]:
+        output_steps = {}
+
+        def get_steps(dct: dict[str, Any]) -> dict[str, Any]:
+            steps = self.get_value('step', dct)
+            if steps is None:
+                return {}
+            times = self.get_value('time', dct)
+            if len(steps) != len(times):
+                self.logger.error(
+                    'Inconsistent step-time combinations in observable data.'
+                )
+
+            return {step: times[n] for n, step in enumerate(steps)}
+
+        def get_observable_steps(
```

Comment:
i suggest you return the output dict here, so you do not redefine output_steps as a list later